ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -140,25 +142,33 @@
   private final KsqlEngine ksqlEngine;
   private final ReplayableCommandQueue replayableCommandQueue;
   private final long distributedCommandResponseTimeout;
+  private final ActiveChecker activeChecker;
+  private final TimeUtils timeUtils;
 
   public KsqlResource(
       final KsqlConfig ksqlConfig,
       final KsqlEngine ksqlEngine,
       final ReplayableCommandQueue replayableCommandQueue,
-      final long distributedCommandResponseTimeout
+      final long distributedCommandResponseTimeout,
+      final ActiveChecker activeChecker
```

Comment:
The code for `KsqlResource` and the other places that report activity can be simplified if you change all the places that report activity so that they just take a simple event interface that they can fire, e.g. a `Runnable` or some specific `Event` interface:

```
@FunctionalInterface
interface Event {
   void fire();
}
```

(NB: see main comments section for why the active queries param is not needed).

The reason this is better is because:
1. It moves the handling of time from being duplicated in every place that calls `onRequest` to a single place: the implementation of the event handler.
1. `KsqlResource` and the others don't need any other method on `ActiveChecker`, passing a simplified event interface decreases coupling.

So the code in `handleKsqlStatements` just becomes:

```
onRequestEvent.fire();
```

Personally, I'd go with:

```suggestion
      final Runnable onRequestEvent
```